libs/langchain/langchain/schema/runnable/utils.py

```diff
@@ -87,12 +87,38 @@ def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef) -> Any:
         IsLocalDict(input_arg_name, self.keys).visit(node)
 
 
+class GetLambdaSource(ast.NodeVisitor):
+    def __init__(self) -> None:
+        self.source: Optional[str] = None
+
+    def visit_Lambda(self, node: ast.Lambda) -> Any:
+        self.source = ast.unparse(node)
+
+
 def get_function_first_arg_dict_keys(func: Callable) -> Optional[List[str]]:
     try:
         code = inspect.getsource(func)
         tree = ast.parse(textwrap.dedent(code))
         visitor = IsFunctionArgDict()
         visitor.visit(tree)
         return list(visitor.keys) if visitor.keys else None
-    except (TypeError, OSError):
+    except (SyntaxError, TypeError, OSError):
+        return None
+
+
+def get_lambda_source(func: Callable) -> Optional[str]:
+    try:
+        code = inspect.getsource(func)
+        tree = ast.parse(textwrap.dedent(code))
+        visitor = GetLambdaSource()
+        visitor.visit(tree)
+        return visitor.source
+    except (SyntaxError, TypeError, OSError):
         return None
+
+
+def indent_lines_after_first(text: str, prefix: str) -> str:
```

Comment:
a bit odd to get a prefix here vs. n_spaces

Reply:
Similar api to textwrap.indent